lib/AllocateDescriptorsPass.cpp

```diff
@@ -924,17 +924,16 @@ bool AllocateDescriptorsPass::CallTreeContainsGlobalBarrier(Function *F) {
         // and StorageBuffer storage classes) semantics are checked because
         // Workgroup variables are inherently coherent (and do not require the
         // decoration).
-        if (call->getCalledFunction()->getName().equals(
-                "__spirv_control_barrier")) {
+        if (call->getCalledFunction()->getName().startswith("spirv.op.224")) {
           // barrier()
-          if (auto *semantics = dyn_cast<ConstantInt>(call->getOperand(2))) {
+          if (auto *semantics = dyn_cast<ConstantInt>(call->getOperand(3))) {
             uses_barrier =
                 semantics->getZExtValue() & kMemorySemanticsUniformMemory;
           }
-        } else if (call->getCalledFunction()->getName().equals(
-                       "__spirv_memory_barrier")) {
+        } else if (call->getCalledFunction()->getName().startswith(
+                       "spirv.op.225")) {
```

Comment:
Need to guard against the case where an instruction number is 2250, for example.
I think this should have a "." after the 225, so: "spirv.op.225."

Reply:
Yes, better safe than sorry.